homeassistant/components/camera/__init__.py

```
@@ -473,6 +492,28 @@ class CameraMjpegStream(CameraView):
             msg['id'], 'image_fetch_failed', 'Unable to fetch image'))
 
 
+@websocket_api.async_response
+@websocket_api.websocket_command({
+    vol.Required('type'): 'camera/stream',
+    vol.Required('entity_id'): cv.entity_id,
+    vol.Optional('format', default='hls'): cv.string,
+})
+async def ws_camera_stream(hass, connection, msg):
+    """Handle get camera stream websocket command.
+
+    Async friendly.
+    """
+    from homeassistant.components.stream import async_request_stream
+    try:
+        fmt = msg['format']
+        camera = _get_camera_from_entity_id(hass, msg['entity_id'])
+        url = await async_request_stream(hass, camera.stream_source, fmt=fmt)
+        connection.send_result(msg['id'], {'url': url})
+    except HomeAssistantError:
+        connection.send_error(
```

log exception, not silent swallow it